### PR TITLE
[REF] [Feature] [56369] Discovery Screen

### PR DESCRIPTION
**Pull request by hungld-1302**
## Related Tickets
- [#Task 56369: Discovery Screen](https://edu-redmine.sun-asterisk.vn/issues/56369)
## WHAT
- Implement Discovery Screen
- Handle loading view
## Evidence (Screenshot or Video)
image
image
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -4,7 +4,26 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:background="@color/color_primary"
     tools:context=".ui.MainActivity">
 
+    <FrameLayout
+        android:id="@+id/fragment_container_home"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
```
Comment (ndt0409) [inline app/src/main/res/layout/activity_main.xml]: extract dimen

File: app/src/main/res/layout/fragment_discovery.xml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        android:layout_height="match_parent"
+        android:padding="@dimen/_20dp">
+
+        <LinearLayout
+            android:id="@+id/layout_search"
+            android:layout_width="match_parent"
+            android:layout_height="@dimen/_50dp"
+            android:background="@drawable/bg_gray_conner_10"
+            android:gravity="center_vertical"
+            android:padding="10dp"
```
Comment (ndt0409) [inline app/src/main/res/layout/fragment_discovery.xml]: .

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -15,8 +15,36 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.TheCocktail"
         tools:targetApi="31">
+        <activity
+            android:name=".ui.detail.DetailActivity"
+            android:exported="false">
```
Comment (hungvv-1301) [inline app/src/main/AndroidManifest.xml]: screenOrientation

File: app/src/main/java/com/sildev/thecocktail/ui/MainActivity.kt [hunk 4]
```diff
@@ -1,13 +1,37 @@
...
+            }
+            true
+        }
+        binding.navHome.setOnItemReselectedListener {  }
+    }
 
+    private fun replaceFragment(fragment: Fragment) {
+        val transaction = fragmentManager.beginTransaction()
+        transaction.replace(R.id.fragment_container_home, fragment)
+        transaction.commit()
```
Comment (ghost) [inline app/src/main/java/com/sildev/thecocktail/ui/MainActivity.kt]: a viết extension nhé

File: app/src/main/java/com/sildev/thecocktail/ui/discovery/DiscoveryFragment.kt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+        binding.apply {
+            recyclerviewIngredients.adapter = ingredientAdapter
+            recyclerviewCocktail.adapter = drinkAdapter
+            textSeeMore.setOnClickListener {
+                startActivity(Intent(context, SeeAllActivity::class.java))
+            }
+            layoutSearch.setOnClickListener {
+                startActivity(Intent(context, SearchActivity::class.java))
+
```
Comment (hungvv-1301) [inline app/src/main/java/com/sildev/thecocktail/ui/discovery/DiscoveryFragment.kt]:
```suggestion

```

File: app/src/main/java/com/sildev/thecocktail/ui/discovery/DiscoveryFragment.kt [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        viewModel.cocktails.observe(viewLifecycleOwner, Observer {
+            drinkAdapter.submitList(it)
+            binding.shimmerCocktails.isVisible = false
+        })
+
+    }
+
+    private fun onClickIngredientItem(ingredient: Ingredient) {
+        val intent = Intent(context, SeeAllActivity::class.java)
+        intent.putExtra(Constant.INGREDIENT_DATA_KEY,ingredient)
```
Comment (hungvv-1301) [inline app/src/main/java/com/sildev/thecocktail/ui/discovery/DiscoveryFragment.kt]:
```suggestion
       EXTRA_
```

File: app/src/main/java/com/sildev/thecocktail/base/BaseViewHolder.kt [hunk 7]
```diff
@@ -4,8 +4,20 @@ import androidx.recyclerview.widget.RecyclerView
 import androidx.viewbinding.ViewBinding
 
 abstract class BaseViewHolder<T, VB : ViewBinding>(
-    open val binding: VB
+    private val binding: VB, click: (T) -> Unit
```
Comment (ghost) [inline app/src/main/java/com/sildev/thecocktail/base/BaseViewHolder.kt]:
```suggestion
    private val binding: VB, click: (T) -> Unit = { _ -> }
```
như này thì sau này có cái nào không cần click thì mình không cần truyền tham số thứ 2 nữa

**Pull request merged**